app/Actions/Photo/Strategies/AddDuplicateStrategy.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Actions\Photo\Strategies;
+
+use App\Exceptions\PhotoSkippedException;
+use App\Models\Logs;
+use App\Models\Photo;
+
+class AddDuplicateStrategy extends AddBaseStrategy
+{
+	public function __construct(AddStrategyParameters $parameters, Photo $existing)
+	{
+		parent::__construct($parameters, $existing);
+	}
+
+	public function do(): Photo
+	{
+		// At least update the existing photo with additional metadata if
+		// available
+		$this->hydrateMetadata();
+		if ($this->photo->isDirty()) {
+			Logs::notice(__METHOD__, __LINE__, 'Updating metadata of existing photo.');
+			$this->photo->save();
+		}
+
+		if ($this->parameters->importMode->shallSkipDuplicates()) {
+			Logs::notice(__METHOD__, __LINE__, 'Skipped upload of existing photo because skipDuplicates is activated');
+			// TODO: Think again of this. A "usual" case should not result in an exception.
+			throw new PhotoSkippedException('This photo has been skipped because it\'s already in your library.');
+		} else {
+			// Duplicate the existing photo, this will also duplicate all
```

Comment:
I'm wondering about the use of a pivot table here.
That would make @nagmat84 happy with a n:m relationship. :)

Reply:
Yes, it would. But this should be another PR.